File: src/main/java/no/nav/arbeidsgiver/min_side/services/unleash/FeatureToggleService.java
package no.nav.arbeidsgiver.min_side.services.unleash;

import no.finn.unleash.Unleash;
import no.finn.unleash.UnleashContext;
import no.nav.arbeidsgiver.min_side.controller.AuthenticatedUserHolder;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Service
public class FeatureToggleService {

    private final Unleash unleash;
    private final AuthenticatedUserHolder authenticatedUserHolder;

    @Autowired
    public FeatureToggleService(Unleash unleash, AuthenticatedUserHolder authenticatedUserHolder) {
        this.unleash = unleash;
        this.authenticatedUserHolder = authenticatedUserHolder;
    }

    public Map<String, Boolean> hentFeatureToggles(List<String> features) {

        return features.stream().collect(Collectors.toMap(
                feature -> feature,
                this::isEnabled
        ));
    }

    public Boolean isEnabled(String feature) {
        return unleash.isEnabled(feature, contextMedInnloggetBruker());
    }

    private UnleashContext contextMedInnloggetBruker() {
        UnleashContext.Builder builder = UnleashContext.builder();
        builder.userId(authenticatedUserHolder.getToken());
        return builder.build();
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/services/unleash/FeatureToggleService.kt
package no.nav.arbeidsgiver.min_side.services.unleash

import no.finn.unleash.Unleash
import no.finn.unleash.UnleashContext
import no.nav.arbeidsgiver.min_side.controller.AuthenticatedUserHolder
import org.springframework.stereotype.Service

@Service
class FeatureToggleService(
    private val unleash: Unleash,
    private val authenticatedUserHolder: AuthenticatedUserHolder
) {
    fun hentFeatureToggles(features: List<String>) =
        features.associateWith { isEnabled(it) }

    fun isEnabled(feature: String) =
        unleash.isEnabled(feature, contextMedInnloggetBruker())

    private fun contextMedInnloggetBruker() =
        UnleashContext.builder().userId(authenticatedUserHolder.token).build()
}